test/integration/serviceaccount/legacy_service_account_token_clean_up_test.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serviceaccount
+
+// This file tests the legacy service account token cleaning-up.
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	clientinformers "k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	serviceaccountcontroller "k8s.io/kubernetes/pkg/controller/serviceaccount"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/serviceaccount"
+	"k8s.io/utils/clock"
+	testingclock "k8s.io/utils/clock/testing"
+)
+
+const (
+	dateFormat    = "2006-01-02"
+	cleanUpPeriod = 24 * time.Hour
+	syncInterval  = 10 * time.Second
+)
+
+func TestLegacyServiceAccountTokenCleanUp(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, kubefeatures.LegacyServiceAccountTokenCleanUp, true)()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	fakeClock := testingclock.NewFakeClock(time.Now().UTC())
+
+	c, config, stopFunc, informers, err := startServiceAccountTestServerAndWaitForCaches(ctx, t)
+	defer stopFunc()
+	if err != nil {
+		t.Fatalf("failed to setup ServiceAccounts server: %v", err)
+	}
+
+	// start legacy service account token cleaner
+	startLegacyServiceAccountTokenCleaner(ctx, c, fakeClock, informers)
+
```

Comment:
is anything waiting for the configmap with the `since` tracking date to be created?

Reply:
Added a function to wait till the configmap is labeled with `since`.